src/context/User.tsx

```diff
@@ -1,9 +1,39 @@
 import { Auth } from '@supabase/ui';
+import { logoutRoutes } from 'app/Unauthenticated';
 import { useClient } from 'hooks/supabase-swr';
+import { useSnackbar } from 'notistack';
+import { useIntl } from 'react-intl';
+import { useNavigate } from 'react-router';
+import { useEffectOnce } from 'react-use';
 import { BaseComponentProps } from 'types';
 
 export const UserProvider = ({ children }: BaseComponentProps) => {
     const supabaseClient = useClient();
+    const navigate = useNavigate();
+    const intl = useIntl();
+    const { enqueueSnackbar } = useSnackbar();
+
+    useEffectOnce(() => {
+        supabaseClient.auth.onAuthStateChange((event) => {
+            if (event === 'PASSWORD_RECOVERY') {
+                enqueueSnackbar(
+                    intl.formatMessage({
+                        id: 'login.passwordReset',
+                    }),
+                    {
+                        anchorOrigin: {
+                            vertical: 'top',
+                            horizontal: 'center',
+                        },
+                        variant: 'info',
+                    }
+                );
```

Comment:
Super weird edge case but might as well catch it. If a user got a recovery email link just adding a notification that they should not need to do that.